server/plugin.go

```diff
@@ -33,9 +35,29 @@ import (
 	"github.com/mattermost/mattermost/server/public/pluginapi/cluster"
 	"github.com/pborman/uuid"
 	"github.com/pkg/errors"
+	"github.com/stretchr/testify/mock"
 	"golang.org/x/oauth2"
 )
 
+var clientMock *mocks.Client = nil
+
+func getClientMock(p *Plugin) *mocks.Client {
```

Comment:
This is the initial mock needed for the plugin to start, and also the one that is going to be used when the environment variable is there.

Reply:
Yes, we can, for the API is a bit more challenging, but we can do it, for sure.